packages/davinci-client/src/lib/davinci.utils.ts

```diff
@@ -56,3 +67,109 @@ export function transformActionRequest(node: NextNode, action: string): DaVinciR
     },
   };
 }
+
+export function handleResponse(cacheEntry: DaVinciCacheEntry, dispatch: Dispatch, status: number) {
```

Comment:
Move all the response handling and error differentiation to this function. This contains quite a bit verbosity, but to keep this readable and more "flat".